graylog2-web-interface/src/components/common/PageSizeSelect.tsx

```diff
@@ -18,61 +18,66 @@ import * as React from 'react';
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
 
-import { Input } from 'components/bootstrap';
+import { DropdownButton, MenuItem } from 'components/bootstrap';
+import { DEFAULT_PAGE_SIZES } from 'hooks/usePaginationQueryParameter';
 
-const Wrapper = styled.div`
-  margin-bottom: 5px;
-
-  && .form-group {
-    margin-bottom: 0;
-  }
-
-  .control-label {
-    padding-top: 0;
-    font-weight: normal;
-  }
-
-  .page-size-select {
-    display: flex;
-    align-items: baseline;
+const StyledDropdownButton = styled(DropdownButton)`
+  ~ .dropdown-menu {
+    min-width: auto;
   }
 `;
 
-const PAGE_SIZES = [10, 50, 100];
+const Container = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 3px;
+`;
 
 type Props = {
   className?: string,
   pageSize: number,
   pageSizes: Array<number>,
-  onChange: (event: React.ChangeEvent<HTMLOptionElement>) => void,
+  onChange: (newPageSize: number) => void,
+  showLabel?: boolean
 };
 
-const PageSizeSelect = ({ pageSizes, pageSize, onChange, className }: Props) => (
-  <Wrapper className={`${className ?? ''} form-inline page-size pull-right`}>
-    <Input id="page-size"
-           type="select"
-           bsSize="small"
-           label="Show"
-           value={pageSize}
-           onChange={onChange}
-           formGroupClassName="page-size-select">
-      {pageSizes.map((size) => <option key={`option-${size}`} value={size}>{size}</option>)}
-    </Input>
-  </Wrapper>
-);
+const PageSizeSelect = ({ pageSizes, pageSize, onChange, className, showLabel }: Props) => {
+  const select = (
+    <StyledDropdownButton className={className}
+                          id="page-size-select"
+                          title={`${pageSize} Rows`}
+                          aria-label="Configure page size"
+                          pullRight
+                          bsSize="small"
+                          bsStyle="default">
+      {pageSizes.map((size) => <MenuItem key={`option-${size}`} onSelect={() => onChange(size)}>{size}</MenuItem>)}
+    </StyledDropdownButton>
+  );
+
+  if (showLabel) {
+    return (
+      <Container className={className}>
+        Show
+        {select}
+      </Container>
+    );
+  }
+
+  return select;
+};
```

Comment:
Maybe it is cleaner to create a separate component to render the `PageSizeSelect` with a label?